src/coreclr/System.Private.CoreLib/src/System/Reflection/Metadata/RuntimeTypeMetadataUpdateHandler.cs

```diff
@@ -11,11 +11,14 @@ namespace System.Reflection.Metadata
     /// <summary>Metadata update handler used to clear a Type's reflection cache in response to a metadata update notification.</summary>
     internal static class RuntimeTypeMetadataUpdateHandler
     {
+        public static bool HotReloadDeltaApplied { get; internal set; }
```

Comment:
We should make this unconditionally return false when hot reload support is stripped.

Reply:
Look for System.Reflection.Metadata.MetadataUpdater.IsSupported in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml for how to do it.

Reply:
> We should make this unconditionally return false when hot reload support is stripped.

Isn't it should also apply to the ClearCache(...) too? Do not clear cache if it is not supported.

Reply:
The ClearCache method should be taken care of by this already: https://github.com/dotnet/runtime/blob/b4991a5a43d379f271e1f26286eb24709e62da9c/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml#L33-L37

When hot reload is not supported, the MetadataUpdateHandlerAttribute will be deleted. It means that nothing is going to be keeping the RuntimeTypeMetadataUpdateHandler alive and the whole thing should be trimmed from the final binary.

Reply:
The HotReloadDeltaApplied is false by default and it is set true only if ClearCache() called, by default it is false (I will update the setter to private), then do we still need to check if hot reload supported each time?

Reply:
Trimming is only able to do something with constants. Static variable is not a constant, even when it is never assigned to. Trimming is not able to figure out that a static variable that is never assigned to is a constant in practice.

Reply:
It helps thanks, I was about to check `MetadataUpdater.IsSupported` each time `HotReloadDeltaApplied` is referenced 😛

Reply:
Update: in offline discussion @jkotas suggested its better to `just check MetadataUpdater.IsSupported. It has larger scope - it will avoid the perf overhead even when trimming is not used (ie majority of our customers)`